CMakeLists.txt

```diff
@@ -0,0 +1,217 @@
+cmake_minimum_required(VERSION 3.7.0)
+
+project(dcos-mesos-modules)
+
+set(MESOS_ROOT "../mesos" CACHE PATH "Mesos root directory.")
+get_filename_component(MESOS_ROOT ${MESOS_ROOT} ABSOLUTE)
+set(BOOST_ROOT_DIR "" CACHE PATH "Boost libs root directory.")
+
+message("MESOS_ROOT is ${MESOS_ROOT}")
+
+# CMAKE MODULE SETUP.
+#####################
+# Paths that are searched when `include(...)` is called.
+list(APPEND CMAKE_MODULE_PATH ${MESOS_ROOT}/cmake)
+list(APPEND CMAKE_MODULE_PATH ${MESOS_ROOT}/3rdparty/cmake)
+list(APPEND CMAKE_MODULE_PATH ${MESOS_ROOT}/src/cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+# Macros.
+include(CTest)
+include(Versions OPTIONAL)
+include(Dependencies)
+
+# Set the default standard to C++11 for all targets.
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+# Do not use, for example, `-std=gnu++11`.
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+add_subdirectory(dockercfg)
+add_subdirectory(logsink)
+add_subdirectory(journald)
```

Comment:
Why do you opt for having separate `CMakeLists.txt` files in these module subdirectories, while others like overlay and metrics have their code inlined in this file?

Reply:
I would prefer to have `CMakeLists.txt` in each directory (per module), however, right now that is not possible because corresponding changes require some changes to source code (that would break autotools build). Probably it makes sense to either add a TODO here or refactor modules layout of a part of autotools removal. thoughts?

Reply:
Please add a TODO here, so that after we remove autotools we don't forget that it was the thing that caused this layout.

Reply:
added.